client/e2e-test/script-test.js

```diff
@@ -0,0 +1,144 @@
+ */
+
+/* global __datasetName */
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/no-unresolved */
+/* eslint-disable no-console */
+
+const puppeteer = require('puppeteer');
+
+const datasetName = Date.now().toString();
+
+(async () => {
+  const browser = await puppeteer.launch({
+    args: [
+      '--no-sandbox',
+      '--disable-setuid-sandbox',
+    ],
+  });
+  const page = await browser.newPage();
+
+  try {
+    // Login
+    console.log('\nSTARTING LUMEN TEST WITH PUPPETEER\n');
+    await page.setViewport({ width: 1024, height: 768 });
+    console.log('Accessing to http://t1.lumen.local:3030...');
+    await page.goto('http://t1.lumen.local:3030/');
+    await page.waitForSelector('#username', { timeout: 10000 });
+    console.log('Typing username...');
+    await page.type('#username', 'jerome');
+    console.log('Typing password...');
+    await page.type('#password', 'password');
+    console.log('Trying login...');
+    await page.click('#kc-login');
+    await page.waitForSelector('button[data-test-id="dataset"]', { timeout: 10000 });
+    console.log('Login was successful.\n');
+    await page.evaluate(`window.__datasetName = "${datasetName}"`);
+
+    // Dataset adding
+    // Click Dataset+ option
+    console.log('Accessing to dataset creation...');
+    await page.click('button[data-test-id="dataset"]');
+    await page.waitForSelector('button[data-test-id="next"]', { timeout: 10000 });
+    // Select link option
+    console.log('Typing dataset link...');
+    await page.click('input[data-test-id="source-option"][value="LINK"]');
+    await page.click('button[data-test-id="next"]');
+    await page.waitForSelector('#linkFileInput', { timeout: 10000 });
+    // Insert link
+    await page.type('#linkFileInput', 'https://github.com/lawlesst/vivo-sample-data/raw/master/data/csv/people.csv');
+    await page.click('button[data-test-id="next"]');
+    await page.waitForSelector('input[data-test-id="dataset-name"]', { timeout: 10000 });
+    // Insert name
+    console.log('Typing dataset name...');
+    await page.type('input[data-test-id="dataset-name"]', datasetName);
+    // Import
+    console.log('Saving dataset...');
+    await page.click('button[data-test-id="next"]');
+    console.log(`Dataset ${datasetName} was successfully created.\n`);
+    await page.waitForSelector(`[data-test-name="${datasetName}"]`);
+
+    // Search of the ID
+    console.log('Extracting dataset ID...');
+    const id = await page.evaluate(() => {
+      const found = document.querySelector(`[data-test-name="${__datasetName}"]`);
+      return Promise.resolve(found.getAttribute('data-test-id'));
+    });
+    console.log(`ID extracted: ${id}\n`);
+    let pending;
+    const timeOut = setTimeout(() => { console.log('Error waiting for pending dataset'); process.exit(1); }, 60 * 1000);
+    do {
+      pending = await page.$(`[data-test-name="${datasetName}"] [data-test-id="pending"]`);
+    } while (pending);
```

Comment:
Does this loop consume 100% cpu? Maybe add a small sleep?

Reply:
Now we added a sleep